libs/iresearch/include/iresearch/analysis/wordnet_synonyms_tokenizer.cpp

```diff
@@ -0,0 +1,172 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2025 SereneDB GmbH, Berlin, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is SereneDB GmbH, Berlin, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include "wordnet_synonyms_tokenizer.hpp"
+
+#include <absl/strings/ascii.h>
+#include <absl/strings/str_replace.h>
+#include <absl/strings/str_split.h>
+#include <re2/re2.h>
+
+#include <iresearch/analysis/pipeline_tokenizer.hpp>
+#include <iresearch/analysis/token_attributes.hpp>
+#include <string_view>
+#include <utility>
+
+namespace irs::analysis {
+
+namespace {
+
+constexpr static size_t kWordnetCountParams = 6;
+
+const RE2 kWordnetPattern(R"(s\(([^)]*)\)\.)");
+
+bool RegexWordnet(const std::string_view s_list, std::string_view* result) {
+  if (s_list.length() <= 2) {
+    return false;
+  }
+
+  re2::StringPiece input(s_list.data(), s_list.size());
```

Comment:
NIT: it's unnecessary re2::StringView is using for std::string_view, so this variable can be removed